[13.0] [MIG] agreement -> base_contract

i-vyshnevska opened the pull request:
migrate agreement

ygol committed:
[MIG] agreement: Migration to 12.0

ygol committed:
[FIX] PR review's remarks

ygol committed:
Delete .#__manifest__.py

ygol committed:
[FIX] remove sql constraint for company_id

ygol committed:
Update agreement.py

ygol committed:
[FIX] add back company_id field

ygol committed:
[FIX] flake8 + demo data

ygol committed:
[ADD] notebook in agreement form

requested by @bealdav

ygol committed:
[UPD] manifest with @max3909 remarks

ygol committed:
[FIX] flake8

ygol committed:
Update __manifest__.py

OCA-git-bot committed:
[UPD] README.rst

bealdav committed:
FIX security issue

bealdav committed:
FIX details

oca-travis committed:
[UPD] Update agreement.pot

rdc02271 committed:
Translated using Weblate (Portuguese (Portugal))

Currently translated at 100.0% (27 of 27 strings)

Translation: contract-12.0/contract-12.0-agreement
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement/pt_PT/

max3903 committed:
[FIX] Missing newline

gurneyalex committed:
[IMP] move some notions from agreement_legal to agreement

we move the is_template field definition and the agreement.type model from
the agreement_legal module to the agreement module.

The fields are not displayed by default, unless the feature is enabled through a
technical feature group, this is configurable in the agreement_sale module (because agreement
in itself has no UI, and agreement_legal enables the feature by default)

oca-travis committed:
[UPD] Update agreement.pot
[UPD] README.rst

Commit:
[13.0][MIG] agreement: migrate + black

max3903 commented:
@i-vyshnevska Please have a look at #386

gurneyalex commented:
@i-vyshnevska  can you rename the module base_contract?

Commit:
[IMP] rename agreement to base_contract

pedrobaeza commented:
Do we finally have an agreement - hehe - on the renaming?

sbejaoui reviewed (changes requested):
Please consider the discussion about 13.0 road-map #386

jgrandguillaume reviewed (approved):
Work well on functional side

guewen commented on base_contract/models/agreement.py:
```diff
@@ -0,0 +1,55 @@
+    company_id = fields.Many2one(
+        "res.company",
+        string="Company",
+        default=lambda self: self.env["res.company"]._company_default_get(),
```
`_company_default_get()` should be removed: https://github.com/odoo/odoo/blob/58492733b81073694a7e2f5e790b31983f91df7d/odoo/addons/base/models/res_company.py#L205

It shows a warning at loading of the module.

```suggestion
        default=lambda self: self.env.company.id,
```

guewen commented on base_contract/views/agreement.xml:
```diff
@@ -0,0 +1,90 @@
+            </div>
+            <group name="main">
+                <group name="left">
+                    <field name="agreement_type_id"
+                           groups="agreement.group_use_agreement_type"/>
+                    <field name="is_template" groups="agreement.group_use_agreement_template"/>
```
Warnings:

```
2020-05-06 13:44:29,775 115 WARNING odoodb odoo.addons.base.models.ir_ui_view: The group agreement.group_use_agreement_type defined in view agreement.form does not exist! 
2020-05-06 13:44:29,776 115 WARNING odoodb odoo.addons.base.models.ir_ui_view: The group agreement.group_use_agreement_template defined in view agreement.form does not exist! 
```

```suggestion
                    <field name="agreement_type_id"
                           groups="base_contract.group_use_agreement_type"/>
                    <field name="is_template" groups="base_contract.group_use_agreement_template"/>
```

I didn't check for other places

alexis-via committed:
FIX ir.rule for odoo 13

Replace deprecated _company_default_get() by standard v13 syntax
Add domain field
Add chatter on agreement object

alexis-via commented:
I fixed and improved several points in this PR via a PR on @i-vyshnevska 's branch https://github.com/i-vyshnevska/contract/pull/1

tbaden commented:
What's the progress on this one? Is there still something todo or does it just need some more reviews?

i-vyshnevska committed:
Merge pull request #1 from akretion/13-mig-agreement-improve

base_contract: Improve your OCA PR

Chabagutor reviewed (approved):
Tested and installed for prodution purpose.

alexis-via commented:
This is ready to be merged. @sbejaoui 's comment is one year old and he asks to take into account the roadmap discussion, and it's precisely what has been implemented here. So let's move on and merge.

alexis-via commented:
There is something I don't understand with "pre-commit run -a" : when I run it on this branch, it doesn't change anything... although Travis complains about black/isort stuff.
If someone with more experience than me on this can sort it out... it's the last thing needed to merge.

rvalyi commented:
@alexis-via probably you should try to rebase this branch on 13.0 before trying your local precommit.
Because from Travis logs it complains about XML formatting from the Prettier tool. But in this 13-mig-agreement, the precommit configuration `.pre-commit-config.yaml` (but may bve also some of the other dot files) is not up to date like in the 13 branch, specially this branch version has no such prettier formating instruction that's why it wont even try to run locally. Instead on Travis it looks it's running the 13.0 pre-commit configuration.

compare:
https://github.com/i-vyshnevska/contract/blob/13-mig-agreement/.pre-commit-config.yaml
https://github.com/OCA/contract/blob/13.0/.pre-commit-config.yaml#L24

Is it clearer now?

alexis-via commented:
A few days ago, I created another PR to replace this one: https://github.com/OCA/contract/pull/578

newtratip commented:
@i-vyshnevska please close this pr supersede https://github.com/OCA/contract/pull/628
